framework/src/base/FEProblemBase.C

```diff
@@ -364,6 +364,8 @@ void FEProblemBase::initialSetup()
   if (_app.isRecovering() && (_app.isUltimateMaster() || _force_restart))
     _resurrector->setRestartFile(_app.getRecoverFileBase());
 
+  _mesh.initialSetup();
```

Comment:
It looks weird to me. This late construction in mesh means that something in mesh depends on other moose objects, like user objects, variables, etc. It would make more sense that this initial setup is after initial condition, uo initial setup. Just some thoughts.